docs/manual/kernel/Vfs.tex

```diff
@@ -0,0 +1,91 @@
+
+\subsection{File Cache}
+The file cache is a separate entity to the VFS. Currently only file content is cached, meaning metadata still requires accessing the backing media.
+
+File contents are cached in chunks of a fixed number of bytes, called a unit. The unit size can be queried at runtime via \verb|FileCacheUnit()|. Files are sparsely cached, meaning only the parts of a file that (rounded to the nearest cache unit) are kept in memory. Non-accessed parts of a file may also be tentatively loaded, but this is not guarenteed.
+
+The file cache is designed with memory mapping of files in mind, and all cache units are created in a way that satisifies the constraints of the MMU. During initialization the unit size is determined based on the data available from \verb|GetHatLimits()|.
+
+Currently cache units must use contiguous physical memory, but there may be scatter-gather support in the future.
+
+\subsection{Usage}
+All the necessary declarations for using the VFS are available in \verb|filesystem/Vfs.h|, global utility functions are available in \verb|filesystem/Filesystem.h|.
+
+Most operations on the VFS follow a similar pattern: find the VFS driver for the target filesystem, find the target node, call a function on the driver passing the node as an argument.
+
+Some operations apply directly to the VFS driver, like mounting and unmounting, or flushing the whole filesystem's cache. These operations don't require a node to be passed, and can be called on the vfs driver directly.
+
+VFS drivers also provided a \verb|Resolve()| function which traverses the VFS starting at the their own root node. This can be used to quickly lookup known files or check for their existence.
+
+Some global utility functions are also provided: \verb|RootFs()| returns the VFS driver for the root filesystem, and \verb|VfsLookup()| is a shortcut for looking up a node based on a path (it can be absolute, or use the working directory of the context).
+
+VFS nodes keep track of the driver that is responsible for them via the \verb|node.driver| field, which can be used to call driver functions. Common operations are also provided as member functions of the VFS node struct. As an example, to read or write a file you can use \verb|node.ReadWrite()| instead of \verb|node.driver.ReadWrite()|.
+
+\subsection{Example Code}
+For the following examples we'll assume you've obtained a handle to target node and it's called \verb|node|. We're also not going to populate any properties of new nodes, but this is something you should do in real usage.
+
+For the following examples we'll use a dummy FsContext called \verb|context|. In real usage this would either be the kernel context (\verb|KernelFsCtxt| or a context created for a specific user or program).
+
+Creating a child node, in this case a file called "childfile.txt":
```

Comment:
not sur why, but the github render "childfile.txt" in red, like something is wrong (maybe you should use \" instead of only "? 

btw i'm not even sure if is an actual problem.

Reply:
I'm not sure why either, it renders fine in the final pdf. Might just be a github issue